### ISSUE TITLE
Use `uuid.Validate` over `uuid.Parse` where appropriate

### ISSUE THREAD

**Issue by nojnhuh** [kubernetes-sigs/cluster-api-provider-azure#4403]
/kind cleanup

**What needs cleanup:**

v1.5.0 of the [uuid](https://pkg.go.dev/github.com/google/uuid) package we use (#4397) introduced a `Validate` function which makes more sense to use than `Parse` when we only care if a string is a valid UUID. Currently, these are the places where we should update to use `Validate`:
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/7f2ccad2846a934906124d75421fb1eb4a58ad25/api/v1beta1/azuremachine_default_test.go#L118
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/7f2ccad2846a934906124d75421fb1eb4a58ad25/api/v1beta1/azuremachine_validation.go#L115
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/7f2ccad2846a934906124d75421fb1eb4a58ad25/exp/api/v1beta1/azuremachinepool_webhook_test.go#L482

**Anything else you would like to add:**
[Miscellaneous information that will assist in solving the issue.]

/help
/good-first-issue

**Comment by dreamjz** [comment]
I'd like to do this.
Should I start after the `uuid` has been upgraded to 1.5.0?

**Comment by nojnhuh** [comment]
Yep, this issue requires #4397 to be merged first which should happen in the next couple of hours. Thank you for picking this up!

/assign @dreamjz

**Comment by kumarankit999** [comment]
I would like to go with this issue, as @dreamjz hasn't done anything yet!!

**Comment by dreamjz** [comment]
Sorry 😢,  I've been busy in the past few weeks and working overtime on weekends. You can solve it.

**Comment by nojnhuh** [comment]
Thank you!
/unassign @dreamjz 
/assign @kumarankit999
